AmplifyPlugins/DataStore/Sources/AWSDataStorePlugin/Core/DataStoreListDecoder.swift

```diff
@@ -69,3 +69,40 @@ public struct DataStoreListDecoder: ModelListDecoder {
         }
     }
 }
+
+public struct DataStoreModelDecoder: ModelProviderDecoder {
+    public static func shouldDecode<ModelType: Model>(modelType: ModelType.Type, decoder: Decoder) -> Bool {
+        guard let json = try? JSONValue(from: decoder) else {
+            return false
+        }
+        // TODO: This needs to be more strict once we have more than one decoder running
+        // without any sort of priority
+        // check if it has the single field
```

Comment:
Test this with DataStore integration tests and identify if there's a race condition on which decoder gets used first, we'll have a problem if the APIPlugin's decoder gets used first and "shouldDecode" is successful, thus returning a lazy model with a model provider of type AppSyncModelProvider instead of DataStoreModelProvider. With that set up, we can figure out how to prioritize DataStore's decoders over API's, such as a sorted list of model decoders based on a priority field.